setup hq marketplace test v2

aguayov opened the pull request.

aguayov committed:
setup hq marketplace test v2

ianbautista commented on database/migrations/1_seed.ts:
```diff
@@ -13,39 +12,35 @@ const products = `
+  (8, 'Snorkel mask', 1000, 1),
+  (9, 'Icee', 75000, 1),
+  (10, 'Sandals', 2000, 1),
+  (11, 'Phone water protector', 6500, 1),
```
Oddly specific. lol

ianbautista commented on src/middlewares/checkoutPayload.middleware.ts:
we are missing the service fee validation here.

ianbautista commented on src/routes/products.routes.ts:
```diff
@@ -0,0 +1,30 @@
+ */
+
+export default function (router: Router, db: any) {
+  router.get('/products', async (_req: Request, res: Response) => {
```
we should do a more structured route. `/vendors/:vendor_id/products` so we aren't hard coding `1` in the query. It'll test them more on API usage.

ianbautista commented on src/utils/queries/getProducts.ts:
```diff
@@ -0,0 +1,8 @@
+const getProducts = () => {
+  return `SELECT p.*
+  FROM products p
+    WHERE p.vendor_id = 1
```
See comment above.

ianbautista commented on src/routes/vendor-fees.routes.ts:
```diff
@@ -0,0 +1,30 @@
+ */
+
+export default function (router: Router, db: any) {
+  router.get('/vendor-fees', async (_req: Request, res: Response) => {
```
this could be `vendors/:vendor_id/fees`

ianbautista commented on src/utils/queries/getVendorFees.ts:
```diff
@@ -0,0 +1,8 @@
+const getVendorFees = () => {
+  return `SELECT *
+    FROM vendor_fees vf
+      WHERE vf.vendor_id = 1
```
see comment in routes.

ianbautista reviewed (changes requested):
Great job on this. Just a few comments.

aguayov committed:
include fees validation

also updated endpoints to use the vendor_id via dynamic routes so that we don't hardcode the values

aguayov committed:
fix route to match naming
